src/psyclone/transformations.py

```diff
@@ -2840,101 +2863,89 @@ def name(self):
 
-        :raises TransformationError: if the target kernel is a built-in.
+        :raises TransformationError: if the node is not a kernel or a routine.
+        :raises TransformationError: if the target is a built-in kernel.
+        :raises TransformationError: if it is a kernel but without an \
+                                     associated PSyIR.
         :raises TransformationError: if any of the symbols in the kernel are \
-                            accessed via a module use statement.
+                                     accessed via a module use statement.
         '''
-        from psyclone.psyGen import BuiltIn
-        if isinstance(kern, BuiltIn):
+        super(ACCRoutineTrans, self).validate(node, options)
+
+        if not isinstance(node, Kern) and not isinstance(node, Routine):
             raise TransformationError(
-                "Applying ACCRoutineTrans to a built-in kernel is not yet "
-                "supported and kernel '{0}' is of type '{1}'".
-                format(kern.name, type(kern)))
+                f"Target of a kernel transformation must be a sub-class of "
+                f"Kern or Routine but got '{type(node).__name__}'")
 
-        # Perform general validation checks. In particular this checks that
-        # the PSyIR of the kernel body can be constructed.
-        super(ACCRoutineTrans, self).validate(kern, options)
+        # If it is a kernel call it must have an accessible implementation
+        if isinstance(node, Kern):
+            if isinstance(node, BuiltIn):
+                raise TransformationError(
+                    f"Applying ACCRoutineTrans to a built-in kernel is not yet"
+                    f" supported and kernel '{node.name}' is of type "
+                    f"'{type(node).__name__}'")
 
-        # Check that the kernel does not access any data or routines via a
-        # module 'use' statement
-        sched = kern.get_kernel_schedule()
-        imported_variables = sched.symbol_table.imported_symbols
-        if imported_variables:
-            raise TransformationError(
-                "The Symbol Table for kernel '{0}' contains the following "
-                "symbol(s) with imported interface: {1}. If these symbols "
-                "represent data then they must first be converted to kernel "
-                "arguments using the KernelImportsToArguments transformation. "
-                "If the symbols represent external routines then PSyclone "
-                "cannot currently transform this kernel for execution on an "
-                "OpenACC device (issue #342).".
-                format(kern.name, [sym.name for sym in imported_variables]))
+            # Get the PSyIR routine from the associated kernel
+            try:
+                kernel_schedule = node.get_kernel_schedule()
+            except GenerationError as error:
+                raise TransformationError(
+                    f"Failed to create PSyIR version of kernel code for "
```

Comment:
s/version//

Reply:
Also, please could you extend the comment to explain why this might fail (i.e. the reason for the try...except).

Reply:
Done both